server/services/src/main/java/org/zanata/service/impl/MachineTranslationServiceImpl.java

```diff
@@ -271,17 +272,29 @@ private MTDocument getTranslationFromMT(MTDocument request,
             log.info("No eligible text flows in document {}", doc.getQualifiedDocId());
             return null;
         }
-
-        MTDocument mtDocument = textFlowsToMTDoc
-                .fromTextFlows(projectSlug, versionSlug,
-                        doc.getDocId(), doc.getSourceLocaleId(),
-                        textFlowsToTranslate,
-                        TextFlowsToMTDoc::extractPluralIfPresent,
-                        backendId);
-        MTDocument result = getTranslationFromMT(mtDocument,
-                targetLocale.getLocaleId());
-        saveTranslationsInBatches(textFlowsToTranslate, result, targetLocale, saveState);
-        return result.getBackendId();
+        int startBatch = 0;
+        String backendIdConfirmation = null;
+        while (startBatch < textFlowsToTranslate.size()) {
+            int batchEnd = Math.min(
+                    startBatch + REQUEST_BATCH_SIZE, textFlowsToTranslate.size());
+            List<HTextFlow> next =
+                    textFlowsToTranslate.subList(startBatch, batchEnd);
+            MTDocument mtDocument = textFlowsToMTDoc
+                    .fromTextFlows(projectSlug, versionSlug,
+                            doc.getDocId(), doc.getSourceLocaleId(),
+                            next,
+                            TextFlowsToMTDoc::extractPluralIfPresent,
+                            backendId);
+            MTDocument result = getTranslationFromMT(mtDocument,
+                    targetLocale.getLocaleId());
+            saveTranslationsInBatches(next, result, targetLocale, saveState);
+            backendIdConfirmation = result.getBackendId();
```

Comment:
// TODO we only return the backendId from the final batch

I think the problem will go away with https://zanata.atlassian.net/browse/ZNTA-2732